source/FreeRTOS_DNS_Callback.c

```diff
@@ -231,7 +231,7 @@
         if( listLIST_IS_EMPTY( &xTempList ) == pdFALSE )
         {
             /* There is at least one item in xTempList which must be removed and deleted. */
-            xEnd = listGET_END_MARKER( &xTempList );
+            xEnd = ( ( const ListItem_t * ) &( xTempList.xListEnd ) );
```

Comment:
Can we use the `listGET_END_MARKER` here, which will enable using mocks to manipulate the `xTempList` during unit testing